ez-launch/sources/ez_launch.move

```diff
@@ -0,0 +1,413 @@
+    ) acquires CollectionDetails, EZLaunchRefs {
+        assert_owner<Collection>(signer::address_of(creator), collection);
+        assert!(vector::length(&token_name_vec) == num_tokens, error::invalid_argument(ETOKEN_METADATA_CONFIGURATION_INVALID));
+        assert!(vector::length(&token_uri_vec) == num_tokens, error::invalid_argument(ETOKEN_METADATA_CONFIGURATION_INVALID));
+        assert!(vector::length(&token_description_vec) == num_tokens, error::invalid_argument(ETOKEN_METADATA_CONFIGURATION_INVALID));
+
+        let object_signer = authorized_collection_signer(creator, collection);
+        let object_signer_address = signer::address_of(&object_signer);
+        let collection_details_obj = borrow_global_mut<CollectionDetails>(object_signer_address);
+        // not check against the ready_to_mint so that we enable creators to pre_mint anytime even when minting has already started
+
+        let i = 0;
+        let length = vector::length(&token_name_vec);
+        while (i < length) {
+            let token = token_pre_mint_internal(
+                creator,
+                collection,
+                *vector::borrow(&token_description_vec, i),
+                *vector::borrow(&token_name_vec, i),
+                *vector::borrow(&token_uri_vec, i),
+            );
+            vector::push_back(&mut collection_details_obj.to_be_minted, token);
+            i = i + 1;
+        };
+    }
+
+    public entry fun mint(
+        user: &signer,
+        collection: Object<Collection>,
+    ) acquires CollectionDetails, EZLaunchRefs {
+        mint_internal(user, collection);
+    }
+
+    public entry fun set_minting_status(creator: &signer, collection: Object<Collection>, ready_to_mint: bool) acquires CollectionDetails, EZLaunchRefs {
+        assert_owner<Collection>(signer::address_of(creator), collection);
+        let object_signer = authorized_collection_signer(creator, collection);
+        let object_signer_address = signer::address_of(&object_signer);
+        let collection_details_obj = borrow_global_mut<CollectionDetails>(object_signer_address);
+
+        collection_details_obj.ready_to_mint = ready_to_mint;
+    }
+        
+    // ================================= View  ================================= //
+
+    #[view]
+    public fun minting_ended(collection: Object<Collection>): bool acquires CollectionDetails, EZLaunchRefs {
+        let object_signer = collection_signer(collection);
+        let object_signer_address = signer::address_of(&object_signer);
+        let collection_details_obj = borrow_global_mut<CollectionDetails>(object_signer_address);
+
+        vector::length(&collection_details_obj.to_be_minted) == 0
+    }
+
+    // ================================= Helper  ================================= //
+
+    fun create_collection_internal(
+        creator: &signer,
+        description: String,
+        name: String,
+        uri: String,
+        mutable_collection_metadata: bool,
```

Comment:
We won't need these also as part of https://github.com/aptos-labs/token-minter/pull/32